### PR TITLE
Coverage

### PR DESCRIPTION
**Pull request by reidkwja**
Updated readme and attached bash script allowing developers to check for coverage % with a terminal command.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,6 @@
+/tests/__pycache__/
+/tuna/__pycache__/
+/tuna/dbBase/__pycache__
+/tuna/utils/__pycache__
```
Comment (alexandraBara) [inline .gitignore:4]: it seems like u added __pycache__ elements to your push. Please remove any __pycache__ folders

File: Jenkinsfile [hunk 2]
```diff
@@ -97,6 +98,16 @@ pipeline {
...
+/*
+        stage("coverage") {
+        agent{  label utils.rocmnode("tunatest") }
+        steps {
+           script {
+           utils.runCodeCovTest()
+           }
+           }
+        }
+*/
```
Comment (alexandraBara) [inline Jenkinsfile:110]: remove?

File: README.md [hunk 3]
```diff
@@ -110,3 +110,12 @@ In order for a PR to be accepted the following `pylint` command needs to result
 cd MITunaX/tuna
 pylint -f parseable -d duplicate-code --max-args=8 --indent-string '  ' *.py
 ```
+## Coverage 
+
+To extract coverage percentage from the underlying code base prior to opening Pull Request:
+
+```
+cd MITunaX/tuna
+bash tests/covscripts/coverage_script.sh
```
Comment (alexandraBara) [inline README.md:119]: make it an executable so we can run with ./tests/conv...

File: tests/covscripts/coverage_script.sh [hunk 4]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
```
Comment (alexandraBara) [inline tests/covscripts/coverage_script.sh:1]: Add copyright to any new files you add including sh scripts

File: tests/covscripts/parse_attributes.py [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import json
```
Comment (alexandraBara) [inline tests/covscripts/parse_attributes.py:1]: same, add copyright

File: tests/covscripts/parse_attributes.py [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+
+coverage_file = open('../MITunaX/tests/covscripts/buffer/coverage.json')
```
Comment (alexandraBara) [inline tests/covscripts/parse_attributes.py:3]: get current path and formulate the path to coverage.json that way. Check out some of our tests/<>.py scripts

File: tests/test_find.py [hunk 7]
```diff
@@ -34,8 +34,8 @@
 
 from tuna.sql import DbCursor
 from tuna.utils.logger import setup_logger
-from tuna.find_db import FindDB
-from tuna.db_tables import ConvConfig, Job
+#from tuna.find_db import FindDBMixin
+#from tuna.db_tables import ConvConfig, Job
```
Comment (alexandraBara) [inline tests/test_find.py:38]: why are these commented out? if pylint is complaining, then remove

File: vars/utils.groovy [hunk 8]
```diff
@@ -428,12 +428,15 @@ def pytestSuite1() {
            sh "pytest tests/test_fin_utils.py"                     
            sh "pytest tests/test_add_session.py"                     
            sh "pytest tests/test_merge_db.py"
+
+
```
Comment (alexandraBara) [inline vars/utils.groovy:432]: remove new lines

File: vars/utils.groovy [hunk 9]
```diff
@@ -504,11 +507,37 @@ def runLint() {
...
+          sh "python3 -m coverage run -m pytest"
+          sh "python3 -m coverage json"
+          sh "mv coverage.json ../MITunaX/tests/covscripts/buffer"
+          sh "python3 tests/covscripts/parse_attributes.py"
+          sh "file= ../MITunaX/tests/covscripts/buffer/coverage_percentage.txt"
+          myvar = sh "\$(cat "$file")"
+          CODE_COV = 10
+          if (CODE_COV > myvar) {
+          error "Not added to env: ${item}"
+          }
```
Comment (alexandraBara) [inline vars/utils.groovy:536]: this seems to be commented out in the JK file. if its not called at all, then remove

**Review by alexandraBara — CHANGES_REQUESTED** [review]

**Comment by alexandraBara** [comment]
@reidkwja  we still need a place to keep track of the % coverage. Did you add a var somewhere that each dev can update per branch with the total % cov? If not, i would suggest adding such a var in: [tests/covscripts/parse_attributes.py]. Something like:
total_code_cov = "10%" and then print it out

File: tests/test_find.py (hunk 7 above)
Comment (reidkwja) [reply tests/test_find.py:38]: I believe test_find requires some modifications, seems like there is a variable name change in find_db script

Comment (alexandraBara) [reply tests/test_find.py:38]: then that should have been fixed by whoever made that change. Please pull in the latest develop, uncomment this and make sure it passes pylint. Our CI indicated this is not an issue.

**Review by alexandraBara — CHANGES_REQUESTED** [review]
The diff still shows 3 new files:
.coverage
.gitignore
.ignore

What are these files?

**Comment by reidkwja** [comment]
> 



> The diff still shows 3 new files: .coverage .gitignore .ignore
> 
> What are these files?

We can delete .gitignore and .ignore

.coverage is automatically generated by the coverage tool.

**Comment by reidkwja** [comment]
@alexandraBara does this require any additional reviews/actions?

**Review by cderb — APPROVED** [review]
LGTM

**Review by alexandraBara — APPROVED** [review]
Please remove the .gitignore file, its still in the diff.

**Pull request merged**